src/main/java/ch/njol/skript/structures/PreloadingStructure.java

```diff
@@ -63,12 +63,13 @@ public abstract boolean init(Expression<?>[] exprs,
 	public void init(SectionNode node) { }
 
 	@Nullable
-	public static PreloadingStructure parse(String expr, SectionNode sectionNode) {
+	public static PreloadingStructure parse(String expr, SectionNode sectionNode, int priority) {
```

Comment:
This design is very limiting in a way that priorities must be distinct in order to work. How can an addon developer check this?

Instead of **comparing** the priorities when parsing, they should be used to order **sort** all the structures. The process should look a bit like this:
1. You sort all matched structures by their priority (higher priority means first in the list). Structure with the same priority are sorted by the occurrence or some other factor (doesn't really matter).
2. You iterate over the sorted list and initialize the structures, without their priority (since we already took care of that)